docs/adr/adr_003_bundling_collection_data.md

```diff
@@ -0,0 +1,230 @@
+ - A bundle of all attachments data per collection (opt-in)
+
+Looking at existing use-cases, we can observe that bundling attachments may not be equally relevant depending on the number of records and their attachment size:
+ - `security-state/intermediates` collection: 1720 records for 3MB in total. Lots of tiny .pem files required as soon as possible after first startup for security features: __bundling very beneficial__
+ - `fakespot-suggest-products` collection: 300 records with a few MB per attachment (eg. 5MB). Since the total size is huge: bundling brings __little advantage__
+ - Machine learning collections: few records with big attachments (eg. 20MB): almost no advantage. 
+ - `Translations-models` collection: 150 records for 1.2GB in total. Clients only download a few pairs when the user enables the feature for certain languages: __bundling is not feasible__
+
+That’s why it would make sense for attachments bundling to be opt-in for stakeholders.
+
+There is already a collection metadata attribute to mark attachments as enabled and/or required. The flag would just consist in adding a new field attachment.bundle: true|false.
+
+Also, since [usage quotas](https://docs.kinto-storage.org/en/stable/api/1.x/quotas.html) are not enabled on the Remote Settings server, consumers can end up publishing huge files and bloat attachments bundles, we may want to have a server side configuration setting to fix a size limit for which the bundles won’t be built. For example, even when opted-in, if the target bundle exceeds 20MB, it won’t be built and published. Clients should thus be smart and fail soft if a bundle is not available.
+
+
+### Bundle Formats
+After some research and testing, we've decided to go with Zip compression. Zip is very easy to implement and allows for good flexibility moving forward. We could change our minds or add other bundle types in the future with very little effort.
+
+- Zip - Reduces text content up to 17%.
+    - Pros: 
+        - Interface available in gecko
+        - Decent compression for our use case
+        - Allows for range requests in the future
+        - Universal format
+    - Cons: 
+        - Not as much compression as lzma.
+        - Double compression if dynamic compression is used
+- Tar - When gzipped (automatically by cdn dynamic compression), we can expect to save around 43% size for text content. Probably not much for other data.
+    - Pros: 
+        - Only bundles, doesn't compress
+        - No CPU used to decompress
+        - Compression done by CDN when using dynamic compression
+    - Cons: 
+        - No existing interface in gecko.
+        - Requires dynamic compression.
+- LZMA (7z) - Reduces text content up to 70%.
+    - Pros: 
+        - Very compressed
+    - Cons: 
+        - More compute intense to compress and decompress
+        - Likely need to ship another library to decompress
+        - Need to decompress after receiving
```

Comment:
We said `Double compression if dynamic compression is used` for Zip

Reply:
Added to this section as well.